src/semester.js

```diff
@@ -1,21 +1,25 @@
 import R from 'ramda'
 import { withinDates } from './date'
+import moment from 'moment-timezone'
 
 const toDate = d => new Date(d)
 
 // FIXME: get rid of Raw functions; raw properties should be handled at system boundaries, in the middleware
 const semesterIntervalRaw = R.props(['startsAt', 'endsAt'])
 const semesterDatesRaw = R.pipe(semesterIntervalRaw, R.map(toDate))
-const dateInSemesterRaw = (semester, date) => withinDates(...semesterDatesRaw(semester), date)
+const dateInSemesterRaw = R.curry((date, semester) =>
+  withinDates(...semesterDatesRaw(semester), date)
+)
 
 const semesterInterval = R.props(['startsOn', 'endsOn'])
 const semesterDates = R.pipe(semesterInterval, R.map(toDate))
-export const dateInSemester = (semester, date) => withinDates(...semesterDates(semester), date)
-
-export function findSemester (semesters, date) {
-  // XXX: must be wrapped since we have a different order
-  const predicate = (sem) => dateInSemesterRaw(sem, date)
+export const dateInSemester = (date, semester) => withinDates(...semesterDates(semester), date)
 
+export function findSemester (facultyId, semesters, date) {
```

Comment:
Does the `facultyId` needs to be parametrised? Can't we just treat it as built-in constant?